### DISCUSSION TITLE
Hide heading characters differently

### DISCUSSION THREAD

**Discussion by chchen2021** [Zettlr/Zettlr#2502] [Ideas]
This is a recap of the discussions in issue [#931](https://github.com/Zettlr/Zettlr/issues/931), where a good screenshot by @SSLPP exists to explain my recommendation for implementing "hide heading characters". Using a second-level heading `## Test` as an example, instead of rendering it to an oddly formatted `h2 Test`, render it as `Test` which is more naturally looking, and make `##` only visible by clicking the formatted title `Test`. Eliminate the intermediate label `h2` completely. 

The reason behind the proposal is that it takes the same amount of work to click `h2` or the actual title`Test` to edit the heading level, but the proposed solution aligns the heading and the body text, making the overall markdown text more natural to read. Since the hashtags are visible by default, as is the case currently, and the hashtag-hiding only occurs by explicit choice, there should not be any confusion over how to edit the hashtags. 

Alternatively, we could consider moving `##` leftward, so that title `Test` is aligned with the body text, but @nathanlesage indicated that such an implementation is difficult with the current CodeMirror implementation.

**Comment by chchen2021** [comment]
@nathanlesage: On a different note: it might be helpful if Discussions and Issues are cross-searchable? I am used to just searching over issues and would not know there is the separate Discussions until you mentioned it in your response. These two categories seem to be numbered consecutively but are invisible to each other in searches.

**Reply by nathanlesage** [reply]
Yes, I have the same feeling. The thing is that GitHub discussions is still new, so they hopefully make it more accessible over time …

**Comment by nathanlesage** [comment]
To summarize my two cents on why Zettlr works the way it does in this regard:

Someone once mentioned that they would like to hide the heading chars, and I've had a look at how other apps do this, and Bear, for instance, does it like we do. However, they indeed made the heading text flush with the body text.

But we might be able to accomodate both, because I just realized that we might potentially make use of the gutter (where the text-collapsing arrows are at) to store the h1, h2, h3, ... indicators. And then just collapse the heading characters completely. Relevant CodeMirror docs: https://codemirror.net/doc/manual.html#setGutterMarker

**Comment by chchen2021** [comment]
I appreciate your responsiveness to suggestions and ideas. 

Whatever the solution is in hiding the heading characters, I think the key is that the rendered title and the body text need to be aligned. Otherwise, we might as well just leave the hashtags in place. 

Before posting here, I tried to find out a way to change the CSS myself to achieve the proposed alignment. However, I had trouble locating a master CSS for Zettlr (on macOS). I tried uncompressing "app.asar" but still could not find it. I am aware of the [Zettlr documentation](https://docs.zettlr.com/en/core/custom-css/) on customizing CSS, but with my limited programming knowledge, it is so much faster if I can look at a master css.

**Reply by nathanlesage** [reply]
Have a look here: https://github.com/Zettlr/Zettlr/blob/develop/source/common/modules/markdown-editor/editor.less

**Reply by chchen2021** [reply]
Thanks for the link to the less file. It's very useful to know where the style files are stored.  It does not look like there is a quick fix, so I hope the heading can be rendered differently in a future update.

**Comment by chchen2021** [comment]
I took a first try at hiding the heading tags with the following custom CSS plus the option of "Preferences | Display | Hide Heading Characters".
```
.heading-tag {
  visibility: hidden;
  margin-left: -2.5rem;
}
```

This CSS is still a bit crude. Changing the heading level can be a pain. I automated the selection/deselection of "Preferences | Display | Hide Heading Characters" with a keyboard shortcut so I can easily show the hashes for editing. Alternatively, one can just delete the old heading and start a new line with the correct heading level. 

To me, the readability is greatly improved by hiding the heading characters. I hope Zettlr eventually implements a more elegant solution.

**Reply by nathanlesage** [reply]
Definitely, I'm more and more thinking about somehow moving them to the gutter, but that requires a little bit more thought.
